docs/adr/0006-consensus-governance.md

```diff
@@ -0,0 +1,459 @@
+  address of each vote is looked up in the prepared entity address mapping. The
+  corresponding number of votes (on the principle of 1 base unit equals one
+  vote) are added to the results mapping based on the voted option. Any votes
+  that are not from the current validator set are ignored and the
+  `invalid_votes` field is incremented for each such vote.
+
+- In case the percentage of votes relative to the total voting power is less
+  than `quorum`, the proposal is rejected.
+
+- In case the percentage of `VoteYes` votes relative to all valid votes is less
+  than `threshold`, the proposal is rejected.
+
+- Otherwise the proposal is passed.
+
+- The proposal's status is changed to either `StatePassed` or `StateRejected`
+  and the proposal is saved.
+
+- The proposal is removed from the list of active proposals.
+
+- The corresponding `ProposalFinalizedEvent` is emitted with the following
+  structure:
+
+  ```golang
+  type ProposalFinalizedEvent struct {
+      // ID is the unique identifier of a proposal.
+      ID uint64 `json:"id"`
+      // State is the new proposal state.
+      State ProposalState `json:"state"`
+  }
+  ```
+
+- In case the proposal has been passed, the deposit is transferred back to the
+  proposal submitter and a corresponding `staking.TransferEvent` is emitted,
+  indicating transfer from the _proposal deposit account_ to the submitter's
+  account.
+
+- In case the proposal has been rejected, the deposit is transferred to the
+  common pool and a corresponding `staking.TransferEvent` is emitted,
+  indicating transfer from the _proposal deposit account_ to the common pool
+  account.
+
+- In case the proposal has been passed, the proposal content is executed.
+
+### Proposal Content Execution
+
+After any proposal is executed the corresponding `ProposalExecutedEvent` is
+emitted with the following structure:
+
+```golang
+type ProposalExecutedEvent struct {
+    // ID is the unique identifier of a proposal.
+    ID uint64 `json:"id"`
+}
+```
+
+#### Upgrade Proposal
+
+The set of pending upgrades is checked to make sure that no upgrades are
+currently pending. If there is an existing upgrade pending the upgrade proposal
+is not executed.
+
+When an upgrade proposal is executed, a new entry is added to the list of
+pending upgrades using `epoch` as `<upgrade-epoch>`.
+
+On each epoch transition (as part of `BeginBlock`) it is checked wheter a
+pending upgrade is scheduled for that epoch. In case it is and we are not
+running the new version, the consensus layer will panic. Otherwise, the pending
+upgrade proposal is removed.
+
+#### Cancel Upgrade Proposal
+
+When a cancel upgrade proposal is executed, the proposal identified by
+`proposal_id` is looked up and removed from the list of pending upgrades. In
+case the pending upgrade does not exist anymore, no action is performed.
+
+### Consensus Parameters
+
+This proposal introduces the following new consensus parameters in the
+governance module:
```

Comment:
Here we can specify: "Since these are consensus parameters, they need to be specified in the genesis file and will be the same across all proposals."

This will help clarify the nature of the parameters.